commit 57e06394dd19191f36880aa8d1d4522a458922ca
Author: Allan McRae <allan@archlinux.org>
Date:   Mon Jun 4 10:33:52 2012 +1000

    makepkg: skip integrity checking early
    
    If "SKIP" is provided for an integrity check, abort checking as soon
    as possible.
    
    Also swap fork to tr for bash4 conversion.
    
    Signed-off-by: Allan McRae <allan@archlinux.org>

diff --git a/scripts/makepkg.sh.in b/scripts/makepkg.sh.in
index 6bf95ff0..26466ab4 100644
--- a/scripts/makepkg.sh.in
+++ b/scripts/makepkg.sh.in
@@ -737,6 +737,12 @@ check_checksums() {
 				file="$(get_filename "$file")"
 				printf "%s" "    $file ... " >&2
 
+				if [[ ${integrity_sums[$idx]} = 'SKIP' ]]; then
+					echo "$(gettext "Skipped")" >&2
+					idx=$((idx + 1))
+					continue
+				fi
+
 				if ! file="$(get_filepath "$file")"; then
 					printf -- "$(gettext "NOT FOUND")\n" >&2
 					errors=1
